### PR TITLE
Context menu for non-text links

### PR DESCRIPTION
**Pull request by psimyn**
Fixes https://github.com/brave/browser-laptop/issues/248

- Check parentNodes to see if the Link context menu items should be shown. 
- Allow showing both link and image menu items (for image-links)
- Show context menu for non-text input types

### CODE REVIEW

File: app/content/webviewPreload.js [hunk 1]
```diff
@@ -156,7 +156,8 @@ ipc.on(messages.SET_AD_DIV_CANDIDATES, function (e, adDivCandidates, placeholder
 })
 
 function hasSelection (node) {
-  if (node && node.selectionStart !== undefined &&
+  if (node && /password|search|tel|text|url/.test(node.type) &&
```
Comment (bridiver) [inline app/content/webviewPreload.js]: I'm not sure if we need this extra check, won't it prevent the copy menu from working on plain text? Is there a case that is broken that this fixes?

**Comment by bridiver** [comment]
Thanks for the PR! This is a much needed fix. Just had one small question on it, but otherwise looks great to me. It would be awesome to get some tests for these.

File: app/content/webviewPreload.js (hunk 1 above)
Comment (psimyn) [inline app/content/webviewPreload.js]: Was specifically for input types. Currently error on input type submit, for example

**Comment by psimyn** [comment]
@bridiver addressed/fixed question - thanks for picking that up!

**Comment by bridiver** [comment]
Thanks! There are currently some conflicts, if you can resolve those and bring it up-to-date I'll get it merged

**Comment by psimyn** [comment]
done

**Comment by bridiver** [comment]
Merged, thanks!

**Pull request merged**